### PR TITLE
Resolve freetype symbols locally

### LINKED ISSUE #58: [Tizen 5.5 watch] Apps cannot launch in candidate mode
.NET type apps crash on my Tizen 5.5 watch device when run in candidate mode (launched from an existing loader process without creating a new process). The issue is probably related to the EFL preloading feature on watch devices. (Removing files in `/usr/share/dotnet.tizen/preload` temporarily fixes the problem - only for rooted devices.)

How to reproduce:
1. Create a project and run on a 5.5 watch device:<p>
   ```
   flutter-tizen create --tizen-language csharp .
   flutter-tizen run
   ```
   The app is run in standalone mode when it's launched for the first time after install.
2. Stop the app.
3. Wait for 5 seconds (until a loader process starts) and launch the app again, by clicking an app icon or using `app_launcher -s`.

lldb caught a segmentation fault at libfreetype.so code section:
```cpp
(lldb) Process 7461 stopped
* thread #21, name = 'io.flutter.ui', stop reason = signal SIGSEGV: invalid address (fault address: 0x30)
    frame #0: 0xec19e0e6 libfreetype.so.6
->  0xec19e0e6: .long  0xb1036b1b                ; unknown opcode
    0xec19e0ea: addshs r4, r6, r8, lsl r7
    0xec19e0ee: .long  0xf8d04770                ; unknown opcode
    0xec19e0f2: push   {r2, r3, r8, r9, r12, sp}
bt
* thread #21, name = 'io.flutter.ui', stop reason = signal SIGSEGV: invalid address (fault address: 0x30)
  * frame #0: 0xec19e0e6 libfreetype.so.6
    frame #1: 0xe4a7e03a libflutter_engine.so
```

Device info:
```sh
bash-3.2# cat /etc/os-release
NAME=Tizen
VERSION="5.5.0 (Tizen5/Unified)"
ID=tizen
VERSION_ID=5.5.0
PRETTY_NAME="Tizen 5.5.0 (Tizen5/Unified)"
ANSI_COLOR="0;36"
CPE_NAME="cpe:/o:tizen:tizen:5.5.0"
BUILD_ID=PLATFORM_DayR800XX_20200611.001
```

The issue doesn't occur on Tizen 6.0 devices because:

- The system freetype library version is newer.
- Apps launch in standalone mode only, because the `http://tizen.org/metadata/direct-launch` metadata value is by default enabled in `tizen-manifest.xml`. The metadata was first introduced in Tizen 6.0.

### PR DESCRIPTION
**Pull request by swift-kim**
- Reland "Roll FreeType to 2.10.4" (reverts #60).
- Provide a version script to the linker to restrict the symbol resolution range. The dynamic linker by default starts to resolve global symbols from loaded shared libraries, not within the current shared library. This results in a reference to undefined symbols on Tizen 5.5 or older devices which load their own version of freetype library onto memory prior to app launch.

Fixes #58.

### CODE REVIEW

File: shell/platform/embedder/flutter_engine_exports.lst [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+# Linker script that hides symbols with the prefix "FT_" so that global freetype
+# symbols referenced by Skia are not overriden by an external shared library at
+# runtime.
+
+{
+  global:
+    *;
+  local:
+    FT_*;
```
Comment (bbrto21) [inline shell/platform/embedder/flutter_engine_exports.lst:13]: First of all, thank you for your hard work. I've reviewed the things that you've tried to solve this problem.
And I'm worried that other libs may also have similar problems. 
I'm not sure about this, what about restrict the global symbols of libflutter_tizen and lib flutter plugins? Will it work?

Comment (bbrto21) [reply shell/platform/embedder/flutter_engine_exports.lst:13]: > `libflutter_tizen.so` or `libflutter_plugins.so` does not have static dependencies on external libs

Um.. I'm worried about some problem related to other libraries statically linked to flutter-engine. Their symbols can also be misresolved on the same principle? not only freetype.

In Android shell, it seems that all symbols are made local and selectively exposed to global. but I'm not sure that is the same purpose that you do. I think you are the most expert on this. I'm just curious about your opinion.

Comment (swift-kim) [reply shell/platform/embedder/flutter_engine_exports.lst:13]: > Their symbols can also be misresolved on the same principle? not only freetype.

Yes, but I don't know the exact list of symbols that `libflutter_engine.so` should export and it's not easy to figure it out.

> In Android shell, it seems that all symbols are made local and selectively exposed to global.

The android shell works very differently from ours. It has a Java wrapper (based on JNI) for all the flutter APIs and therefore has to only export `JNI_OnLoad`.

Comment (bbrto21) [reply shell/platform/embedder/flutter_engine_exports.lst:13]: All right, let's move on as it is

**Review by bbrto21 — APPROVED** [review]
LGTM

**Pull request merged**